### ISSUE TITLE
What's the parameter setting used in paper?

### ISSUE THREAD

**Issue by JiuTongBro** [buaacyw/IT3D-text-to-3D#5]
Hi, thanks for your attractive work firstly.

I am trying to re-implement your experiments. However, it seems the parameter setting released in your code is quite different from that used in the paper.

The first difference I noticed is that, in Sec. 5.1 of the paper, you mentioned you trained the base models for about 25k steps, and refined them for around 20k steps. But in your code realization. the refining takes about 20k epochs, which is 20k*100(opt.dataset_size_train) steps, which is much larger than the setting in paper.

Another difference is that, according to your GAN weight decay setting in the code, where opt.g_loss_decay_begin_step=25000 and opt.g_loss_decay_step=2500, the GAN loss will decay rapidly, but not gradually during the refinement. I drew a plot about the refine epochs(x-axis, it is epochs, not steps) and the losses(y-axis), as you can see, the GAN loss fastly reaches zero according to your current setting.
![image](https://github.com/buaacyw/IT3D-text-to-3D/assets/56216579/e5485e74-0ac1-4d92-b548-437328cbc5ef)

_**As a result, the refined models have no much difference compared to the base model (as there is almost no GAN constraints).**_

I wonder is there any misunderstanding? Or could you please provide the official parameter setting used in paper?

Thanks a lot!

**Comment by buaacyw** [comment]
See https://github.com/buaacyw/IT3D-text-to-3D/blob/3793c5fab54985ce25b7e183819f246ffab97d18/main.py#L486
This is just a dummy implementation. In GAN mode, I do this to tune the best hyperparameter. In my experiment I will terminate the training manually and check the wandb log.
You can delete this line and it will be the same as the paper setting.

**Comment by buaacyw** [comment]
For the second problem, the plot is correct. I think you misunderstand the hyperparameter meaning.
For example, if we resume from 20K step and the `g_loss_decay_begin_step` is set to 25000 and `g_loss_decay_step` is set to 2500. The GAN weight is `g_loss_weight` from 20K step to 25K step and decays to `g_loss_weight_end` from 25K step to 27.5K step.

**Comment by zyhrainbow** [comment]
> the GAN loss will decay rapidly, but not gradually during the refinement. I drew a plot about the refine epochs(x-axis, it is epochs, not steps) and the losses(y-axis), as you can see, the GAN loss fastly reaches zero according to your current setting.

I have the same problem，but I still can not understand correctly. Can you explain more? Thanks~

**Comment by zyhrainbow** [comment]
> For the second problem, the plot is correct. I think you misunderstand the hyperparameter meaning. Check this function:
> 
> https://github.com/buaacyw/IT3D-text-to-3D/blob/3793c5fab54985ce25b7e183819f246ffab97d18/nerf/utils.py#L2176-L2183
> 
> 
> For example, if we resume from 20K step and the `g_loss_decay_begin_step` is set to 25000 and `g_loss_decay_step` is set to 2500. The GAN weight is `g_loss_weight` from 20K step to 25K step and decays to `g_loss_weight_end` from 25K step to 27.5K step. So, the GAN constraints have been applied for 7.5K step and you would find a better result in wandb log around this step. The best tuning step differs depending on the complexity of the coarse model. You can also tune it for 20K step and would find a slightly better result for most prompts.

I have the same problem，but I still can not understand correctly. Can you explain more? Thanks~

**Comment by buaacyw** [comment]
> 

Hi, it's quite simple: the training is manually killed after the GAN training ends.
